client/src/unifyfs_api_internal.h

```diff
@@ -37,6 +37,7 @@ typedef struct {
     int pending_unlink;           /* received unlink callback */
 
     int needs_sync;               /* have unsynced writes */
```

Comment:
For clarity, I think we should rename these as:
`needs_sync` -> `needs_write_sync`
`needs_extents_sync` -> `needs_read_sync`

Reply:
needs_read_sync feels like read data is synchronizing. Maybe we should call it needs_read_metadata/extents_sync

Reply:
did the change